streaming/src/main/scala/org/apache/gearpump/streaming/appmaster/DagManager.scala

```diff
@@ -0,0 +1,104 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.gearpump.streaming.appmaster
+
+import akka.actor.{ActorRef, Actor}
+import org.apache.gearpump.cluster.UserConfig
+import org.apache.gearpump.partitioner.{LifeTime, PartitionerDescription}
+import org.apache.gearpump.streaming.appmaster.DagManager.{DAGScheduled, WatchChange, DAGOperationSuccess, DAGOperationFailed, DAGReplace, TaskLaunchData, LatestDAG, GetTaskLaunchData, GetLatestDAG}
+import org.apache.gearpump.streaming.task.{TaskId, Subscriber}
+import org.apache.gearpump.streaming.{ProcessorId, StreamApplication, ProcessorDescription, DAG}
+import org.apache.gearpump.util.Graph
+
+/**
+ * Will handle dag modification and other stuff related with DAG
+ * @param userConfig
+ */
+
+class DagManager(userConfig: UserConfig) extends Actor {
+  implicit val system = context.system
+
+  private var dags = List(DAG(userConfig.getValue[Graph[ProcessorDescription,
+        PartitionerDescription]](StreamApplication.DAG).get))
+
+  private var watcher: Option[ActorRef] = None
+
+  private def taskLaunchData(dag: DAG, processorId: Int, context: AnyRef): TaskLaunchData = {
+    val processorDescription = dag.processors(processorId)
+    val subscribers = Subscriber.of(processorId, dag)
+    TaskLaunchData(processorDescription, subscribers, context)
+  }
+
+  def receive: Receive = {
+    case GetLatestDAG =>
+      sender ! LatestDAG(dags.last)
+    case GetTaskLaunchData(version, processorId, context) =>
+      dags.find(_.version == version).foreach { dag =>
+        sender ! taskLaunchData(dag, processorId, context)
+      }
+    case DAGReplace(oldProcessorId, newProcessor) =>
+      if (dags.length > 1) {
```

Comment:
why need to check dags.length>1 here. My understanding is dags has all the history instead of waiting-for-handling queue.